### PR TITLE
Disk plugin: UdevNameAttr feature improvements

### PR DESCRIPTION
**Pull request by rpv-tomsk**
disk plugin: Initialize udev context only once, not per each read
disk plugin: Call disk_udev_attr_name() only when UdevNameAttr configured

### CODE REVIEW

**Comment by rubenk** [comment]
Thanks @rpv-tomsk!

File: src/disk.c [hunk 1]
```diff
@@ -867,9 +867,13 @@ static int disk_read (void)
...
 #if HAVE_LIBUDEV
-		char *alt_name = disk_udev_attr_name (handle_udev, disk_name, conf_udev_name_attr);
-		if (alt_name != NULL)
-			output_name = alt_name;
+		char *alt_name = NULL;
+		if (conf_udev_name_attr != NULL)
+		{
+			alt_name = disk_udev_attr_name (handle_udev, disk_name, conf_udev_name_attr);
+			if (alt_name != NULL)
+				output_name = alt_name;
```
Comment (rubenk) [inline src/disk.c]: This leaks alt_name if ignorelist_match is true and we continue the loop.
disk_udev_attr_name allocates, so the result should be freed.

Comment (tokkee) [inline src/disk.c]: Fixed in #1741

Comment (rpv-tomsk) [inline src/disk.c]: Hmm, I need to upgrade some my machines now...

Thank you for catching this.

Comment (rubenk) [inline src/disk.c]: Thank Coverity ;) I should have caught it during review, sorry.

Comment (rpv-tomsk) [inline src/disk.c]: The problem caused by #1565.  First version of #1565 was even worse, some of problems was fixed in new revision, but this one was overlooked.

Comment (rubenk) [inline src/disk.c]: Ah well. At the end it is an improvement, and this is why we have a master branch.
Let me know if you'd like access to Coverity btw.

**Pull request merged**